sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/ModelHelper.java

```diff
@@ -15,8 +18,18 @@
  * RESERVED FOR INTERNAL USE.
  */
 public class ModelHelper {
-    public static final Pattern IP_V4_URL_PATTERN = Pattern
-        .compile("(?:\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3})|(?:localhost)");
+
+    /**
+     * Determines whether or not the passed authority is IP style, that is it is of the format
+     * "&lt;host&gt;:&lt;port&gt;".
+     *
+     * @param authority The authority of a URL.
+     * @throws MalformedURLException If the authority is malformed.
+     * @return Whether the authority is IP style.
+     */
+    public static boolean determineAuthorityIsIpStyle(String authority) throws MalformedURLException {
+        return new URL("http://" +  authority + "/path").getPort() != -1;
```

Comment:
Any reason `/path` is required?

Reply:
It was a required parameter for an overload of URL constructor that I was initially trying to use, but I can try this without it and see if it still works